src/process_util.rs

```diff
@@ -97,6 +101,66 @@ pub fn run_and_wait(cmd: &mut Command)
     unreachable!();
 }
 
+pub fn run_with_tty_and_wait(cmd: &mut Command, tty_fd: i32)
+    -> Result<i32, String>
+{
+    let pgid = unsafe { getpgrp() };
+    let child = match cmd.spawn() {
+        Ok(child) => child,
+        Err(e) => return Err(format!("Error running {:?}: {}", cmd, e)),
+    };
+    let mut trap = Trap::trap(&[SIGCHLD, SIGTERM]);
+    'signal_loop: for signal in trap.by_ref() {
+        match signal {
+            SIGCHLD => {
+                debug!("Received SIGCHLD signal, waiting child");
+                loop {
+                    match waitpid(child.pid(), Some(WNOHANG | WUNTRACED)) {
```

Comment:
You wait only on single child. But other children may die too (even if we have not created them). This code pobably leaves zombie processes behind.